### ISSUE TITLE
Intermittent failure in db::tests::lock_tracker_metrics

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#1627]
__Steps to reproduce:__
Run CI checks enough time

__Expected behaviour:__
Tests pass always

__Actual behaviour:__
`db::tests::lock_tracker_metrics` test intermittently fails

here is an example failure on PR https://github.com/influxdata/influxdb_iox/pull/1620 that has no code changes (only docs): https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/6544/workflows/e3c8b185-ca3f-4c63-9b58-27af3d5195fa/jobs/26335

```

---- db::tests::lock_tracker_metrics stdout ----
thread 'db::tests::lock_tracker_metrics' panicked at 'called `Result::unwrap()` on an `Err` value: FailedMetricAssertionError { name: "catalog_lock_wait_seconds_total", msg: "0.042846025 > 0.07 failed", metrics: "# HELP catalog_chunks_mem_usage_bytes Memory usage by catalog chunks\n# TYPE catalog_chunks_mem_usage_bytes gauge\ncatalog_chunks_mem_usage_bytes{db_name=\"lock_tracker\",source=\"mutable_buffer\",svr_id=\"10\"} 44\ncatalog_chunks_mem_usage_bytes{db_name=\"lock_tracker\",source=\"parquet\",svr_id=\"10\"} 0\ncatalog_chunks_mem_usage_bytes{db_name=\"lock_tracker\",source=\"read_buffer\",svr_id=\"10\"} 0\n# HELP catalog_chunks_total In-memory chunks created in various life-cycle stages\n# TYPE catalog_chunks_total counter\ncatalog_chunks_total{db_name=\"lock_tracker\",state=\"open\",svr_id=\"10\"} 1\n# HELP catalog_lock_total number of times the tracked locks have been obtained\n# TYPE catalog_lock_total counter\ncatalog_lock_total{access=\"exclusive\",db_name=\"lock_tracker\",lock=\"partition\",svr_id=\"10\"} 1\ncatalog_lock_total{access=\"shared\",db_name=\"lock_tracker\",lock=\"partition\",svr_id=\"10\"} 2\ncatalog_lock_total{access=\"exclusive\",db_name=\"lock_tracker\",lock=\"chunk\",partition=\"1970-01-01T00\",svr_id=\"10\"} 2\ncatalog_lock_total{access=\"shared\",db_name=\"lock_tracker\",lock=\"chunk\",partition=\"1970-01-01T00\",svr_id=\"10\"} 1\n# HELP catalog_lock_wait_seconds_total time spent waiting to acquire any of the tracked locks\n# TYPE catalog_lock_wait_seconds_total counter\ncatalog_lock_wait_seconds_total{access=\"exclusive\",db_name=\"lock_tracker\",lock=\"partition\",svr_id=\"10\"} 0\ncatalog_lock_wait_seconds_total{access=\"shared\",db_name=\"lock_tracker\",lock=\"partition\",svr_id=\"10\"} 0\ncatalog_lock_wait_seconds_total{access=\"exclusive\",db_name=\"lock_tracker\",lock=\"chunk\",partition=\"1970-01-01T00\",svr_id=\"10\"} 0\ncatalog_lock_wait_seconds_total{access=\"shared\",db_name=\"lock_tracker\",lock=\"chunk\",partition=\"1970-01-01T00\",svr_id=\"10\"} 0.042846025\n" }', server/src/db.rs:2866:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by alamb** [comment]
@tustvold  did you fix this one?

**Comment by alamb** [comment]
I haven't seen it for a long time

**Comment by tustvold** [comment]
@crepererum did in https://github.com/influxdata/influxdb_iox/commit/27b9477a

**Comment by crepererum** [comment]
I think it wasn't fixed, see #1784.
